tests/aws/services/stepfunctions/v2/callback/test_callback.py

```diff
@@ -513,7 +501,7 @@ def test_start_execution_sync_delegate_timeout(
         )
 
     @markers.snapshot.skip_snapshot_verify(paths=["$..MD5OfMessageBody"])
```

Comment:
Validation against AWS works now ✅ 

However, I am getting different checksums when validating against AWS (the same for the other test):
<img width="1320" alt="Screenshot 2024-04-17 at 12 31 31" src="https://github.com/localstack/localstack/assets/8161292/a15e18f3-7887-4dc8-b6ec-317b0528eb16">

Are you sure that the checksum is deterministic on AWS? Is it the checksum of a static body or does it include some dynamic fields (e.g., datetime)?

If the checksum is not deterministic against AWS, we should change it into a transformer.

Reply:
You have noticed this discrepancy in snapshot tests for time now, however have never made it up the priority to be addressed yet. Here stepFunctions is merely forwarding the SQS reply to the rest of the program. I believe more focus should go in verifying the behaviour of the SQS provider, and secondly, like you say, verify if the message body should be recomputed by stepfunctions and account for non deterministic behaviours. 

I am adding an item in the backlog to investigate this aspect

Reply:
Maybe it's a misunderstanding. My comment is about the **test case** and not about LocalStack.
If the checksum changes every time we re-create the snapshot, it should be transformed (e.g., that's the case if the checksum depends on some dynamic field such as datetime). If AWS always produces the same deterministic snapshot for the checksum (e.g., the checksum depends on some static field and NEVER changes), then the current approach is fine.

Hence, if we create the snapshot (`SNAPSHOT_UPDATE=1`) and re-run the test against AWS without updating the snapshot (`SNAPSHOT_UPDATE=0`), there should be no difference. I'm getting a different checksum for multiple runs against AWS. Can you reproduce this?

Reply:
Thank you Joel, I understand your point now (btw sorry there was a typo in my previous message). I will definitely look into this!

Reply:
After some investigations, it appears as though in StepFunctions, the MD5OfMessageBody is not a direct representation of the message body only. So this changes for every run.
I have added an sqs transformer utility for stepfunctions which treats MD5OfMessageBody, and I have updated all of the snapshots affected.

Reply:
great 👏 Thank you for addressing this 👍 


Now, we have a clear separation between 🚀 :
* snapshot_skip => LocalStack parity issue
* transformer => dynamic behavior, also in AWS